### PR TITLE
Prepare 0.21.1 release

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit prepares for the 0.21.1 release by bumping the version
strings to reflect the new version. This is just a bugfix release that
fixes bugs since the 0.21.0 release.

### Details and comments

TODO:

- [x] Wait for pending backports
- [x] Update release notes

### CODE REVIEW

File: releasenotes/notes/0.21/remove-hard-time-limit-vf2-be83830ecc71f72c.yaml [hunk 1]
```diff
@@ -1,25 +0,0 @@
----
-upgrade:
```
Comment (kevinhartman) [inline releasenotes/notes/0.21/remove-hard-time-limit-vf2-be83830ecc71f72c.yaml:2]: Did the associated changeset get reverted?

File: releasenotes/notes/decompose-fix-993f7242eaa69407.yaml [hunk 2]
```diff
@@ -1,4 +1,7 @@
 ---
 fixes:
-  - Fixed a bug in :meth:`.QuantumCircuit.decompose` that caused the
-    `gates_to_decompose` argument to be handled incorrectly.
\ No newline at end of file
+  - Fixed an issue in :meth:`.QuantumCircuit.decompose` method when
+    passing in a list of ``Gate`` classes for the ``gates_to_decompose``
+    argument. If any gates in the circuit had a label set this argument
+    wouldn't be handled correctly and cause the output decomposition to
```
Comment (kevinhartman) [inline releasenotes/notes/decompose-fix-993f7242eaa69407.yaml:6]:
```suggestion
    wouldn't be handled correctly and caused the output decomposition to
```

File: releasenotes/notes/0.21/remove-hard-time-limit-vf2-be83830ecc71f72c.yaml (hunk 1 above)
Comment (mtreinish) [reply releasenotes/notes/0.21/remove-hard-time-limit-vf2-be83830ecc71f72c.yaml:2]: This is actually just a git artifact. This was part of 0.21.0 not 0.21.1, but I accidentally committed it to the wrong path (see #8320) so it wasn't picked up as part of 0.21.0. The git history shows the file being created after 0.21.0. I just removed the note to ensure it wasn't picked up as part of 0.21.1. But for completeness I should add this to a note file already in 0.21.0. Let me do that really quickly

Comment (mtreinish) [reply releasenotes/notes/0.21/remove-hard-time-limit-vf2-be83830ecc71f72c.yaml:2]: Done in: https://github.com/Qiskit/qiskit-terra/pull/8398/commits/4fb4391e096106a41a9c531112d4dbde1bbdbe55

**Review by kevinhartman — APPROVED** [review]

**Pull request merged**